pkg/operator/render.go

```diff
@@ -51,51 +53,33 @@ func toYAML(i interface{}) []byte {
 	return out
 }
 
-type installConfigGetter func() (installertypes.InstallConfig, error)
-
-func discoverMCOConfig(f installConfigGetter) (*mcfgv1.MCOConfig, error) {
-	ic, err := f()
-	if err != nil {
-		return nil, err
+func createDiscoveredControllerConfigSpec(infra *configv1.Infrastructure, network *configv1.Network) (*mcfgv1.ControllerConfigSpec, error) {
+	if len(network.Spec.ServiceNetwork) == 0 {
+		return nil, fmt.Errorf("service cidr is empty in Network")
 	}
-	dnsIP, err := clusterDNSIP(ic.Networking.ServiceCIDR.String())
+	dnsIP, err := clusterDNSIP(network.Spec.ServiceNetwork[0])
 	if err != nil {
 		return nil, err
 	}
 
-	platform, err := platformFromInstallConfig(ic)
-	if err != nil {
-		glog.Warningf("Warning: %v, using %s", err, platform)
+	platform := "none"
+	switch infra.Status.Platform {
```

Comment:
wondering why "none" isn't the default case?

Reply:
because that would make it
```go
var platform string
switch infra.Status.Platform {
case
case
default:
  platform = "none"
}
```
not sure if ^^ is more cleaner.

Reply:
yeah probably not.